syntax = "proto3"; // 版本声明，使用Protocol Buffers v3版本
import "google/protobuf/empty.proto";
option go_package = "./rpc";  // 指定生成的Go代码在你项目中的导入路径

package rpc; // 包名


// 定义服务
service Greeter {
    rpc WritePoints (WritePointsRequest) returns (WritePointsResponse) {}
    rpc QuerySeries (QuerySeriesRequest) returns (QuerySeriesResponse) {}
    rpc Config (ConfigRequest) returns (ConfigResponse) {}
    rpc QueryRange (QueryRangeRequest) returns (QueryRangeResponse) {}
    rpc QueryTagValues (QueryTagValuesRequest) returns (QueryTagValuesResponse) {}
    rpc QueryNewPoint (QueryNewPointRequest) returns (QueryNewPointResponse) {}
    rpc QuerySeriesAllData (QuerySeriesAllDataRequest) returns (QuerySeriesAllDataResponse) {}
    rpc QueryAllData (google.protobuf.Empty) returns (QueryAllDataResponse) {}
}

// 请求消息
message WritePointsRequest {
    string row = 1;
}
// 响应消息
message WritePointsResponse {
    string reply = 2;
}
// 请求消息
message QuerySeriesRequest {
    string tags = 3;
}

// 响应消息
message QuerySeriesResponse {
    string reply = 4;
}
message ConfigRequest {
    string name = 11;
}
message ConfigResponse {
    string reply =6;
}


// 请求消息
message QueryRangeRequest {
    string metric_tags = 7;
}

// 响应消息
message QueryRangeResponse {
    string reply = 8;
}

// 请求消息
message QueryTagValuesRequest {
    string tag = 9;
}

// 响应消息
message QueryTagValuesResponse {
    string reply = 10;
}
// 请求消息
message QueryNewPointRequest {
    string tag = 9;
}

// 响应消息
message QueryNewPointResponse {
    string reply = 10;
}

// 请求消息
message QuerySeriesAllDataRequest {
    string metric_tags = 7;
}
// 响应消息
message QuerySeriesAllDataResponse {
    string reply = 2;
}
// 响应消息
message QueryAllDataResponse {
    string reply = 12;
}